pkg/controller/common.go

```diff
@@ -470,10 +471,9 @@ func (c *SharedState) getVolumesForPodObjs(pods ...v1.Pod) ([]string, error) {
 	return volumes, nil
 }
 
-func (c *SharedState) addPod(pod *v1.Pod, updateOption updateWithLock) {
-
+func addPod(ctx context.Context, azr azReconciler, pod *v1.Pod, updateOption updateWithLock) error {
```

Comment:
Why drop the receiver? As a standalone function, `addPod` is less meaningful because it no longer reflects that it updates the `SharedState`. If the motivation was to give access to the AzDisk client, then perhaps it should be part of the `SharedState` as well, enabling the controllers to share the cache state?